uobtheatre/productions/models.py

```diff
@@ -622,13 +624,33 @@ def sales_breakdown(self, breakdowns: list[str] = None):
         """Generates a breakdown of the sales of this performance"""
         from uobtheatre.bookings.models import Booking
 
-        return Payment.objects.filter(
+        return Transaction.objects.filter(
             pay_object_id__in=self.bookings.values_list("id", flat=True),
             pay_object_type=ContentType.objects.get_for_model(Booking),
         ).annotate_sales_breakdown(  # type: ignore
             breakdowns
         )
 
+    def refund_bookings(self, authorizing_user: User, send_admin_email=True):
+        """Refund the performance's bookings"""
+        if not self.disabled:
+            raise CantBeRefundedException(f"{self} is not set to disabled")
+
+        refunded_bookings = []
+        for booking in self.bookings.filter(status=Payable.PayableStatus.PAID):
+            if not booking.can_be_refunded:
+                continue
+            booking.refund(authorizing_user=authorizing_user, send_admin_email=False)
```

Comment:
It would be nice if the email told us what happened. I.e. give the following
refunded_bookings
failed_bookings
skipped_bookings